### DISCUSSION TITLE
Hi, is there a way to use this in Platformio without Arduino platform?

### DISCUSSION THREAD

**Discussion by shablam** [Bodmer/TFT_eSPI#2101] [General]
HI, i want to use this but in Platformio i am using wizio-pico. framework is bare metal. Below are the basic flags for ref. 

I try to add the header <TFT_eSPI.h> and it causes lot of wrrors in the header file and elsewhere such as "variable "uint16_t" is not a type name".
it looks for the definition of uint16_t in the tft_espi.h file rather than the _stndin file. This is in C so not sure if there is a C++ / C issue. Maybe you can help. 

`
platform = wizio-pico
board = raspberry-pi-pico
framework = baremetal
build_flags = 
	
	-D PICO_STDIO_UART
	-D PICO_USB_HOST
`

**Comment by Bodmer** [comment]
Yes, users have done this. You must load the Arduino support package into PlatformIO, there are online tutorials for this I I do not provide support for that aspect..

**Reply by shablam** [reply]
Sorry for putting it as an issue, I have not used github posting a lot i usually just figure things out. 

OK thanks. I have Arduino in there but having a hard time with it accepting certain words in the header file especially like namespace. I think it has something to do with gcc ang g++ not interpreting the header as C== as opposed to C, hence why namespace is not accepted.   I am pretty advanced in some aspect of PIO but not this linker flags. I am not sure if its only looking at C or also C++. grr. ok thanks. 

I did add this extra_script but not sure if the flags are right. apparently for gcc if you add g++ you need -x other wise everything get treated as g++. 

`
Import("env")

#
# Dump build environment (for debug)
# print(env.Dump())
#

env.Append(
  LINKFLAGS=[
      "-x",
      "-g++"
  ]
)
`

if anyone has any thoughts let me know.

**Reply by Bodmer** [reply]
The content of a typical platformio.ini configuration file that folk use is [here](https://github.com/Bodmer/TFT_eSPI/blob/master/docs/PlatformIO/Configuring%20options.txt).

**Reply by shablam** [reply]
looks like he has some examples that he adds manually. its the wizio framework and the tft_espi files that clash. I use TinyUSB and those files are not having any issues. 
https://github.com/Wiz-IO/wizio-pico-examples/tree/main/baremetal/pico-ILI9341

**Reply by Bodmer** [reply]
I am unable to help. The library runs fine under the Arduino framework. So maybe avoid using the baremetal framework and drop the use of WizIO.

**Reply by shablam** [reply]
ya no worries. I am way past Arduino. I did bare metal because TinyUSB does not need it. Wizio Pico is very needed. It is the only framework that will connect a Pico with Platformio and the Pico SDK. So no way would I change code I spent months making. I will find a way. Its only some small conflicts right now. thanks for your time! I thought someone else may post their thoughts if they had any. Appreciate your time.
